pulp_2to3_migration/app/pre_migration.py

```diff
@@ -76,29 +77,24 @@ def pre_migrate_content(content_model, mutable_type, lazy_type, premigrate_hook)
         mutable_type: Boolean that indicates whether the content type is mutable.
     """
     batch_size = 500
-    content_type = content_model.pulp2.TYPE_ID
     pulp2content = []
     pulp2mutatedcontent = []
+    content_type = content_model.pulp2.TYPE_ID
+    set_pulp2_repo = content_model.pulp_2to3_detail.set_pulp2_repo
 
     # the latest timestamp we have in the migration tool Pulp2Content table for this content type
     content_qs = Pulp2Content.objects.filter(pulp2_content_type_id=content_type)
     last_updated = content_qs.aggregate(Max('pulp2_last_updated'))['pulp2_last_updated__max'] or 0
     _logger.debug('The latest migrated {type} content has {timestamp} timestamp.'.format(
         type=content_type,
         timestamp=last_updated))
-    set_pulp2_repo = content_model.pulp_2to3_detail.set_pulp2_repo
 
     query_args = {}
-
-    if not set_pulp2_repo:
-        # query only newly created/updated items
-        query_args["_last_updated__gte"] = last_updated
```

Comment:
Problem A:  We only want to save ```Pulp2Content``` for new content (```_last_updated__gte=last_updated```)
Problem B:  But if  ```set_pulp2_repo``` is true for the content type, then we do want to save new ```Pulp2Content``` depending on whether the content was added to a new repo (referencing the RepoContent table)

Solution:  Make a completely separate loop to handle Problem B instead of trying to do both in the same piece of code.